build/rewrite.go

```diff
@@ -54,17 +57,73 @@ func allowedSort(name string) bool {
 	return false
 }
 
-// Rewrite applies the high-level Buildifier rewrites to f, modifying it in place.
+// Rewrites controls the rewrites to be applied.
+//
+// If non-nil, the rewrites with the specified names will be run. If
+// nil, a default set of rewrites will be used that is determined by
+// the type (BUILD vs default starlark) of the file being rewritten.
+type Rewriter struct {
+	DisableRewrites                 *[]string
+	LabelArg                        *map[string]bool
+	Denylist                        *map[string]bool
+	SortableListArg                 *map[string]bool
+	SortDenyList                    *map[string]bool
+	SortAllowList                   *map[string]bool
+	NamePriority                    *map[string]int
+	StripLabelLeadingSlashes        *bool
+	ShortenAbsoluteLabelsToRelative *bool
+}
+
 func Rewrite(f *File) {
+	var rewriter = &Rewriter{
+		LabelArg:                        &tables.IsLabelArg,
+		Denylist:                        &tables.LabelDenylist,
+		SortableListArg:                 &tables.IsSortableListArg,
+		SortDenyList:                    &tables.SortableDenylist,
+		SortAllowList:                   &tables.SortableAllowlist,
+		NamePriority:                    &tables.NamePriority,
+		StripLabelLeadingSlashes:        &tables.StripLabelLeadingSlashes,
+		ShortenAbsoluteLabelsToRelative: &tables.ShortenAbsoluteLabelsToRelative,
+	}
+	rewriter.Rewrite(f)
+}
+
+func (w *Rewriter) Rewrite(f *File) {
+	initRewriter(w)
 	for _, r := range rewrites {
-		if !disabled(r.name) {
-			if f.Type&r.scope != 0 {
-				r.fn(f)
-			}
+		if !disabled(w, r.name) && (f.Type&r.scope != 0) || (w.DisableRewrites != nil) {
+			r.fn(f, w)
 		}
 	}
 }
 
+func initRewriter(w *Rewriter) {
+	if w.LabelArg == nil {
```

Comment:
Doing this has the side effect of changing the passed-in Rewriter object, which would be an unexpected side-effect. I would suggest passing along the nil values as they are now, and creating local variables where the values are used.